common/persistence/execution_manager.go

```diff
@@ -1119,6 +1178,27 @@ func (m *executionManagerImpl) toWorkflowMutableState(internState *InternalWorkf
 	return state, nil
 }
 
+func (m *executionManagerImpl) assertAndConvertArchetypeID(
+	archetypeID chasm.ArchetypeID,
+	methodName string,
+) (chasm.ArchetypeID, bool) {
+	if !softassert.That(
+		m.logger,
+		archetypeID != chasm.UnspecifiedArchetypeID,
+		"ArchetypeID not specified, defaulting to Workflow.",
+		tag.Operation(methodName),
+		tag.SysStackTrace(string(debug.Stack())),
+	) {
+		return chasm.WorkflowArchetypeID, true
+	}
+
+	if archetypeID == chasm.SchedulerArchetypeID {
```

Comment:
Put a comment here that this is because schedulers have a temporary exception to allow them run in the same ID space as workflows.

Reply:
Done.